### PR TITLE
[Feature] Test sample RayCluster YAMLs to catch invalid or out of date ones

### LINKED ISSUE #642: Test sample RayCluster YAMLs to catch invalid or out of date ones
There are lots of sample RayCluster YAMLs in this repo. They are at risk of falling out of date with CRD schemas or best practices. Testing the sample YAMLs will help.

[motivation from this issue](https://github.com/ray-project/kuberay/pull/501#issuecomment-1233452960)

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?
Use #605 to test sample RayCluster YAMLs. This PR found:
* Invalid YAML: #675 
* Outdated YAML: [ray-cluster.getting-started.yaml](https://github.com/ray-project/kuberay/pull/678/files#diff-3d64c6c76d352d559a59d272a4520231692ec59eb903ce98b67cb9411bb075f2)
* Bug: #679

## Related issue number
Closes #642 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

```bash
cd tests/framework
python3 test_sample_raycluster_yamls.py 2>&1 | tee log
```

Screen Shot 2022-11-03 at 4 29 53 PM

### CODE REVIEW

File: ray-operator/README.md
Comment (DmitriGekhtman) [inline ray-operator/README.md:87]:
```suggestion
# Create a RayCluster and a ConfigMap with hello world Ray code.
```

**Comment by DmitriGekhtman** [comment]
Will lgtm once you get the new test to pass consistently.

**Comment by kevin85421** [comment]
@DmitriGekhtman There are 4 tests hit the CPU constraint ("a standard Linux runner has 2-core CPU (x86_64), 7GB of RAM, and 14GB of SSD space.") of GitHub Actions as we discussed in the [Design: KubeRay E2E Configuration tests](https://docs.google.com/document/d/1OyMBgQ9zx5FF0GV3f3WKVdW-Y4OXz_epaywYgTw893w/edit#).

I used `os.system(f'kubectl describe pods -n={self.namespace}')` to gather more information when `RayClusterAddCREvent` fails to converge. You can search "Insufficient cpu" in [CI result](https://github.com/ray-project/kuberay/actions/runs/3391806145/jobs/5638179569).

```
Events:
  Type     Reason            Age               From               Message
  ----     ------            ----              ----               -------
  Warning  FailedScheduling  8s (x3 over 92s)  default-scheduler  0/1 nodes are available: 1 Insufficient cpu.
```

**Comment by DmitriGekhtman** [comment]
Ok, it seems that to automate test execution, we will have to run these tests in the Ray CI.

Here's a proposed sequence of actions.

For this PR, make sure the tests are passing manually when you run them.

After merging this PR, the main priority is manual release testing:

The Ray 2.1.0 release is in progress and a more-or-less final rayproject/ray:2.1.0 image is available.
Run the tests with the 2.1.0 images manually and make sure they pass.
(It is possible that some minor commits will be picked in and the image will be modified before release... just ignore that for now. Or if you like repeat the tests when the Ray team is approaching the final 2.1.0 release -- should be within the next few days; you can check the relevant Anyscale-internal slack channel for details.)

Prior to the KubeRay 0.4.0 release, run the tests manually again with a KubeRay 0.4.0 candidate and Ray 2.1.0.

The next priority is establishing automated pipelines for running these tests in the Ray CI. It's important to track this and do it eventually, but it's fine if we don't put a deadline on it right now.

**Comment by DmitriGekhtman** [comment]
If there's a way to fit some subset of the test into CI, perhaps with modifications to the configs, that would also be good.

**Comment by kevin85421** [comment]
> If there's a way to fit some subset of the test into CI, perhaps with modifications to the configs, that would also be good.

This PR runs 3 tests on KubeRay CI, and I will open issue #695 to track the progress of running tests on Ray CI.

Although each standard Linux runner has 2 cores, I always failed to schedule Pods when CPU usage is higher than 800m (0.8 CPU). Maybe there are some CPU fragmentation problems.

**Review by DmitriGekhtman — APPROVED** [review]
Looks good!

raycluster.complete.yaml and raycluster.autoscaler.yaml are important ones to test in the Ray CI.

**Comment by davidxia** [comment]
Thanks, just wondering if this would've caught the bug fixed by https://github.com/ray-project/kuberay/pull/501?

**Comment by DmitriGekhtman** [comment]
Probably not in its current form, but it wouldn't be too hard to extend the framework to validate log volume mounts.

**Pull request merged**
